🚧Version 1.0 Roadmap

rayapetkova opened the pull request.

TamerAnefiev committed:
added main file

karastoyanov committed:
Dir structure Update

karastoyanov committed:
Dir Structure Update

TamerAnefiev committed:
added base plane image

TamerAnefiev committed:
init

TamerAnefiev committed:
added test file for movements and fps

karastoyanov committed:
project structure update
 - new directory images added
 - requirements file added

TamerAnefiev committed:
updated the path of the image

TamerAnefiev committed:
init

karastoyanov committed:
Main Menu created
 - basic layout structure
 - gold and rank layouts created
 - main buttons layout created
 - custom images and fonts added

karastoyanov committed:
main menu update

TamerAnefiev committed:
init

TamerAnefiev committed:
Merge remote-tracking branch 'origin/development' into development

TamerAnefiev committed:
added unfinished abstract plane

rayapetkova committed:
Added a plane_menu and the functionality for "Start Game"

- Added a plane_menu (fonts need to be implemented, colors need to be changed and information for each plane)
- When the button "Start game" is clicked it calls the function start_game_button_func. Then the function start_plane_menu_window() is called from the choose_a_plane_menu file and the new window is starting. When this happens the main window hides.
- The back button needs to be implemented

karastoyanov committed:
major update:
 - additional graphical objects added / follow the dir structure
 - main menu background music added

karastoyanov committed:
images and sounds added

rayapetkova committed:
Update

rayapetkova committed:
Merge branch 'development' of https://github.com/Stake-And-Rope/py-planes into development

karastoyanov committed:
separate effect for planes buttons

karastoyanov committed:
Minor Changes
 - settings menu added
 - QMediaPlayer integrated with playlist and loopback

karastoyanov committed:
gitignore update

rayapetkova committed:
Update

rayapetkova committed:
Made two new layouts, stylesheet changed

- Added a top layout for the gold icon and gold value
- Added a layout between the top and bottom layout. It is for the title "Plane's info". Added "layout_for_plane_info_name.addSpacing(-25)" which makes the layout closer to the bottom_layout

karastoyanov committed:
background images added

TamerAnefiev committed:
added json handler class for the user settings

TamerAnefiev committed:
updated the structure of the file and renamed the file

TamerAnefiev committed:
Merge branch 'development' of https://github.com/Stake-And-Rope/py-planes into development

TamerAnefiev committed:
added helpers

TamerAnefiev committed:
added convert_alpha to the loaded image

TamerAnefiev committed:
made greenPlane class

TamerAnefiev committed:
deleted old file

karastoyanov committed:
new images

karastoyanov committed:
Merge branch 'development' of github.com:Stake-And-Rope/py-planes into development

karastoyanov committed:
exec rules mod

karastoyanov committed:
new images added

karastoyanov committed:
qss fixed

rayapetkova committed:
Added a functionality for changing the QGroupBox when needed

- When the button plane_button is clicked it calls a function open_plane_info which receives the name of the current plane. When we go to the function open_plane_info we use return lambda for the next function.
- Made a new file display_info in which we will make the new groupboxes
- When we return lambda: plane_info_display(plane_name) we hide the plane_info_groupbox and call the function from the other file (gui\display_plane_info). It gives us the new groupbox and we change the current groupbox with the new one

karastoyanov committed:
json settings added

karastoyanov committed:
settings menu reworked

karastoyanov committed:
username textbox added in settings file

TamerAnefiev committed:
update

TamerAnefiev committed:
update

TamerAnefiev committed:
Merge branch 'development' of https://github.com/Stake-And-Rope/py-planes into development

karastoyanov committed:
Merge branch 'development' of github.com:Stake-And-Rope/py-planes into development

karastoyanov commented on game/planes/green_plane.py:
```diff
@@ -6,4 +8,4 @@ class GreenPlane(BasePlane):
                  fly_speed: float):
-        super().__init__(model, fly_speed)
\ No newline at end of file
+        super().__init__(model, fly_speed)
```
* Test codereview

karastoyanov reviewed (changes requested):
* Check the test code review

TamerAnefiev committed:
test

karastoyanov committed:
settings file update

TamerAnefiev committed:
update with functions

karastoyanov committed:
Merge branch 'development' of github.com:Stake-And-Rope/py-planes into development

karastoyanov committed:
read settings major fix

karastoyanov committed:
read settings major fix

TamerAnefiev committed:
update

TamerAnefiev committed:
added functions to read and overwrite files

TamerAnefiev committed:
removed useless return

karastoyanov committed:
settings menu reading from json

TamerAnefiev committed:
fixed back slash with forward slash file paths

karastoyanov committed:
Merge branch 'development' of github.com:Stake-And-Rope/py-planes into development

karastoyanov committed:
settings overwrite functionality implemented

karastoyanov committed:
main menu background added

karastoyanov committed:
files relationships created

karastoyanov committed:
planes dipslay info fixed

karastoyanov committed:
stylesheet update

karastoyanov committed:
user bullet image

TamerAnefiev committed:
test

TamerAnefiev committed:
update

karastoyanov committed:
stylesheets update

TamerAnefiev committed:
deleted unused files

TamerAnefiev committed:
Merge branch 'development' of https://github.com/Stake-And-Rope/py-planes into development

TamerAnefiev committed:
added borders and added extra checks for the spawn point

TamerAnefiev committed:
added three functions

TamerAnefiev committed:
added changes based on the changes of other files

TamerAnefiev committed:
added background loop functionality

karastoyanov committed:
button stylesheets applied in all qt menus

karastoyanov committed:
settings background added

TamerAnefiev committed:
added bullets logic

TamerAnefiev committed:
added bullet class

TamerAnefiev committed:
update event loop

TamerAnefiev committed:
update middle screen property

karastoyanov committed:
planes menu stylesheets update

karastoyanov committed:
Merge branch 'development' of github.com:Stake-And-Rope/py-planes into development

karastoyanov committed:
README file created

karastoyanov committed:
Rename README.md.md to README.md

karastoyanov committed:
README update

karastoyanov committed:
README update

karastoyanov committed:
Update README.md

karastoyanov committed:
credits menu implemented

TamerAnefiev committed:
added weapon locations for the bullets

TamerAnefiev committed:
removed hardcoded methods

TamerAnefiev committed:
added get bullet speed function

TamerAnefiev committed:
fixed order of imports which caused pygame display error

karastoyanov committed:
Merge branch 'development' of github.com:Stake-And-Rope/py-planes into development

karastoyanov committed:
updated main menu music

karastoyanov committed:
main menu music disabled

karastoyanov committed:
new settings info message

karastoyanov committed:
planes menu buttons update

karastoyanov committed:
enemy bullet added

karastoyanov committed:
enemy planes rotated 180 degrees

TamerAnefiev committed:
updated imports

TamerAnefiev committed:
init

TamerAnefiev committed:
added enemy bullet

TamerAnefiev committed:
added user bullet

TamerAnefiev committed:
updated imports

karastoyanov committed:
tower image added

karastoyanov committed:
music settings updated

TamerAnefiev committed:
converted to abstract class

TamerAnefiev committed:
added enemy plane class

TamerAnefiev committed:
added user plane class

TamerAnefiev committed:
changed imports

TamerAnefiev committed:
added rect attribute

TamerAnefiev committed:
added abstract property and added class attributes

TamerAnefiev committed:
added random moving to left and right and increased shoot cooldown

TamerAnefiev committed:
overwrited get_shoot_cd property

TamerAnefiev committed:
added random enemy plane X coordinates function

TamerAnefiev committed:
update so we can visually see how the enemy plane is moving

rayapetkova committed:
Update health and armor bar

rayapetkova committed:
Changed some of the attributes' names and added an if condition or the methods "reduce_health_bar' and 'reduce_armor_bar'

TamerAnefiev committed:
now accepting loaded images

TamerAnefiev committed:
class which controls all the enemies

TamerAnefiev committed:
update init

TamerAnefiev committed:
added random image getter

TamerAnefiev committed:
added the enemy controller calls

TamerAnefiev committed:
added tower bullet class

TamerAnefiev committed:
deleted some abstract methods

TamerAnefiev committed:
added logic for the towers

TamerAnefiev committed:
update

TamerAnefiev committed:
added tower class

TamerAnefiev committed:
update

TamerAnefiev committed:
included the logic for the tower functionallity

rayapetkova committed:
Logic for rects (needs to be corrected)

rayapetkova committed:
Implemented a FloatRect class

rayapetkova committed:
Bar logic

rayapetkova committed:
Bar logic (changed some attributes' names)

rayapetkova committed:
Bar logic fixed

- Added a method for increasing the current_value
- Added a method that returns the new_width of the bar (the maths part)
- We use the method 'lower_upper_bar_width' in the first method 'draw_bar' so we don't make all the calculations there

rayapetkova committed:
Bars' drawing logic changed

- When the hide_bar_when_empty is False the bar will not be hidden when it is empty. If you set the optional argument as True the bar will hide when it becomes empty.

rayapetkova committed:
Collision logic and masks

TamerAnefiev committed:
added TODO comments, since i hardcoded some values

TamerAnefiev committed:
changed order of drawing

TamerAnefiev committed:
added two new methods and set plane stats

TamerAnefiev committed:
now creating health and armor bars on every enemy spawn. Added bar drawing in the update enemy methods

TamerAnefiev committed:
added sticky health and armor bars

TamerAnefiev committed:
added sticky health and armor bars

TamerAnefiev committed:
update init/super

TamerAnefiev committed:
changed color of armor bar

TamerAnefiev committed:
update

TamerAnefiev committed:
added logic for extra bullets powerup

TamerAnefiev committed:
added extra bullets powerup

TamerAnefiev committed:
added class which controls the powerups

TamerAnefiev committed:
update for the powerup

karastoyanov committed:
armor/health bar update

rayapetkova committed:
Collision fix

rayapetkova committed:
Update

rayapetkova committed:
Collision update

rayapetkova committed:
Collision with another enemy updated

rayapetkova committed:
New colour fills the window when there is collision between user_plane and enemy

rayapetkova committed:
Update

rayapetkova committed:
When the user collides with enemy the game ends
